### ISSUE TITLE
Documentation on custom rule context reporter is misleading, messageId is not supported in reportTranslator.js

### ISSUE THREAD

**Issue by iamgollum** [eslint/eslint#11694]
**Tell us about your environment**

* **ESLint Version:** 4.12.0
* **Node Version:** 8.9.1
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**

eslint ... Linter() 

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "root": true,
  "env": {
    "node": true,
    "browser": true,
    "es6": true,
    "mocha": true
  },
  "plugins": ["import", "prettier", "mocha", "json"],
  "parserOptions": {
    "ecmaVersion": 2017,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "extends": ["eslint:recommended", "plugin:import/errors", "plugin:import/warnings"],
  "rules": {
    "no-console": "off",
    "no-unused-vars": ["warn", { "vars": "all", "args": "none" }],
    "global-require": "error",
    "no-new-require": "error",
    "no-path-concat": "error",
    "no-var": "error",
    "no-process-exit": "off",
    "linebreak-style": ["off", "unix"],
    "no-extra-semi": "off",
    "no-mixed-spaces-and-tabs": "off",
    "import/no-named-as-default": "off",
    "prettier/prettier": "error",
    "mocha/no-exclusive-tests": "error"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
module.exports = {
  meta: {
    messages: {
      avoidName: "Avoid using variables named '{{ name }}'"
    }
  },
  /*  
  create (function) returns an object with methods that ESLint calls to “visit” nodes while traversing the abstract syntax tree (AST as defined by ESTree) of JavaScript code:
if a key is a node type or a selector, ESLint calls that visitor function while going down the tree
if a key is a node type or a selector plus :exit, ESLint calls that visitor function while going up the tree
if a key is an event name, ESLint calls that handler function for code path analysis
A rule can use the current node and its surrounding tree to report or fix problems.
  */
  create: function(context) {
    return {
      Identifier(node) {
        if (node.name === 'foo') {
          context.report({
            node,
            messageId: 'avoidName', // 
            data: {
              name: 'foo'
            }
          });
        }
      }
    };
  }
};

// OTHER FILE

const { Linter, RuleTester } = require('eslint');

const linter = new Linter();
const rule = require('./cfmRule.eslint');

linter.defineParser('my-custom-parser', {
  parse(code, options) {
    console.log(code);
  }
});

linter.defineRule('cfmUsage', rule);

// { parser: 'my-custom-parser' }

const results = linter.verify('const var foo = a;');
console.log(results);

// THROWS ERROR
const ruleTester = new RuleTester();
ruleTester.run('my-rule', rule, {
  valid: ['bar', 'baz'],
  invalid: [
    {
      code: 'foo',
      errors: [
        {
          messageId: 'avoidName'
        }
      ]
    }
  ]
});

```

```bash

```

**What did you expect to happen?**

The test should have passed

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/aaron.tobias@ibm.com/workspace/hertz/hre/node_modules/eslint/lib/report-translator.js:106
    return descriptor.message.replace(/\{\{\s*([^{}]+?)\s*\}\}/g, (fullMatch, term) => {
                              ^

TypeError: Cannot read property 'replace' of undefined
```

The report-translator is expecting a `message` yet the documentation: https://eslint.org/docs/developer-guide/working-with-rules hints that it supports `messageId` instead.

**Are you willing to submit a pull request to fix this bug?**

Yes ... depending on response.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. I think `messageId` is only supported starting in ESLint v4.15.0 -- if you update to a more recent version of ESLint, the rules should work as expected.

**Comment by iamgollum** [comment]
The documentation saids >= 3.0.0

**Comment by not-an-aardvark** [comment]
Perhaps the documentation could be more clear -- it describes a new rule format that was introduced in v3.0.0, but additional features have been gradually added to the rule format after v3.0.0 was released.

v4.12.0 is over a year old, so I'm not sure why you're getting that version when you run `npm install`. You might want to check your `package.json` file and/or your lockfile to make sure you aren't explicitly requiring v4.12.0 to be installed.

**Comment by iamgollum** [comment]
If the documentation can be updated, that would be phenomenal.
Thank you sir!
